token/services/db/common/checks.go

```diff
@@ -204,3 +209,59 @@ func (a *DefaultCheckers) checkUnspentTokens(context context.Context) ([]string,
 	}
 	return errorMessages, nil
 }
+
+func (a *DefaultCheckers) checkTokenSpendability(context context.Context) ([]string, error) {
+	var errorMessages []string
+
```

Comment:
not really important but it would make more sense to have the TMS, network, tokenVault, queryEngine etc as fields of the checker, since we know the tmsID. That spares us the search every time.

Reply:
good point, will do.

Reply:
makes sense. Can you add a comment in the code please so that it's clearly defined why we don't make this optimization?

Reply:
will do